### PR TITLE
AEA-1387 Allow to explicitly declare behaviours that are registered programmatically. 

### LINKED ISSUE #1480: Allow to explicitly declare behaviours that are registered programmatically.
**Is your feature request related to a problem? Please describe.**
I am getting this warning message for an one shot behaviour that I want to register from then handler.
`Class DataUpdateBehaviour of type behaviours found but not declared in the configuration file ../packages/aris/skills/rail_receiver/behaviours.py.`

**Describe the solution you'd like**

I would like to have the option to declare the behaviour in the skill.yaml but to be able to call it when I want from code.

**Describe alternatives you've considered**
None

**Additional context**
None.

### PR DESCRIPTION
**Pull request by marcofavorito**
## Proposed changes

Two changes:
- Reintroduce warning messages when a class not declared in config is found in a skill module (2f83384).
  - This was removed after #2319
- Allow to explicitly declare behaviours that are registered programmatically. (828c702, e29a98d)

## Fixes

Fix #1480

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

n/a

### CODE REVIEW

File: aea/skills/base.py [hunk 1]
```diff
@@ -756,13 +782,13 @@ def _print_warning_message_for_non_declared_skill_components(
...
-    skill_path: str,
+    module_path: str,
 ) -> None:
     """Print a warning message if a skill component is not declared in the config files."""
     for class_name in classes.difference(config_components):
         skill_context.logger.warning(
-            "Class {} of type {} found but not declared in the configuration file {}.".format(
-                class_name, item_type, skill_path
+            "Class {} of type {} found in skill module {} but not declared in the configuration file.".format(
+                class_name, item_type, module_path
```
Comment (marcofavorito) [inline aea/skills/base.py:791]: I think `skill_path` was a wrong name

File: aea/skills/base.py [hunk 2]
```diff
@@ -877,15 +896,16 @@ def _filter_classes(
...
+
         :param classes: a list of pairs (class name, class object)
         :return: a list of the same kind, but filtered with only skill component classes.
         """
         filtered_classes = filter(
-            lambda name_and_class: any(
-                re.match(component, name_and_class[0])
-                for component in self._all_component_names
-            )
-            and issubclass(name_and_class[1], SkillComponent)
```
Comment (marcofavorito) [inline aea/skills/base.py:888]: if we apply this filter, we wouldn't get any class that is not declared, and so no warning message

File: tests/test_skills/test_base.py [hunk 3]
```diff
@@ -480,10 +486,10 @@ def test_print_warning_message_for_non_declared_skill_components():
             "path",
         )
         mock_logger_warning.assert_any_call(
-            "Class unknown_class_1 of type type found but not declared in the configuration file path."
+            "Class unknown_class_1 of type type found in skill module path but not declared in the configuration file."
         )
         mock_logger_warning.assert_any_call(
-            "Class unknown_class_2 of type type found but not declared in the configuration file path."
+            "Class unknown_class_2 of type type found in skill module path but not declared in the configuration file."
```
Comment (marcofavorito) [inline tests/test_skills/test_base.py:492]: due to "refactoring" of the message

File: aea/skills/base.py [hunk 4]
```diff
@@ -1099,8 +1119,53 @@ def _match_class_and_configurations(
...
+                    ).is_programmatically_defined
+                ):
+                    continue
+                _print_warning_message_for_non_declared_skill_components(
+                    self.skill_context,
+                    {unused_class.__name__},
+                    set(),
+                    self._type_to_str(component_type_class),
+                    str(path),
+                )
```
Comment (marcofavorito) [inline aea/skills/base.py:1167]: this is just a way to reuse old code...

**Review by solarw — APPROVED** [review]
LGTM

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Pull request merged**